src/mlpack/methods/ann/rnn_impl.hpp

```diff
@@ -61,6 +63,32 @@ RNN<OutputLayerType, InitializationRuleType, CustomLayers...>::~RNN()
   }
 }
 
+template<typename OutputLayerType, typename InitializationRuleType,
+         typename... CustomLayers>
+void RNN<OutputLayerType, InitializationRuleType, CustomLayers...>::
+  CheckInputDim(size_t numRows, const char *functionName)
+{
+  for (size_t l = 0; l < network.size(); ++l)
+  {
+    size_t layerInSize = boost::apply_visitor(InSizeVisitor(), network[l]);
+    if (layerInSize != 0){
+      if (layerInSize != numRows){
```

Comment:
You could also combine these two `if`s: `if (layerInSize != 0 && layerInSize != numRows)`.  That might make the code cleaner.